cmd/node/metrics/metrics.go

```diff
@@ -122,19 +131,67 @@ func registerPollConnectedPeers(
 	networkComponents *factory.Network,
 ) error {
 
-	numOfConnectedPeersHandlerFunc := func(appStatusHandler core.AppStatusHandler) {
-		numOfConnectedPeers := uint64(len(networkComponents.NetMessenger.ConnectedAddresses()))
-		appStatusHandler.SetUInt64Value(core.MetricNumConnectedPeers, numOfConnectedPeers)
+	p2pMetricsHandlerFunc := func(appStatusHandler core.AppStatusHandler) {
+		computeNumConnectedPeers(appStatusHandler, networkComponents)
+		computeConnectedPeers(appStatusHandler, networkComponents)
 	}
 
-	err := appStatusPollingHandler.RegisterPollingFunc(numOfConnectedPeersHandlerFunc)
+	err := appStatusPollingHandler.RegisterPollingFunc(p2pMetricsHandlerFunc)
 	if err != nil {
 		return errors.New("cannot register handler func for num of connected peers")
 	}
 
 	return nil
 }
 
+func computeNumConnectedPeers(
+	appStatusHandler core.AppStatusHandler,
+	networkComponents *factory.Network,
+) {
+	numOfConnectedPeers := uint64(len(networkComponents.NetMessenger.ConnectedAddresses()))
+	appStatusHandler.SetUInt64Value(core.MetricNumConnectedPeers, numOfConnectedPeers)
+}
+
+func computeConnectedPeers(
+	appStatusHandler core.AppStatusHandler,
+	networkComponents *factory.Network,
+) {
+	peersInfo := networkComponents.NetMessenger.GetConnectedPeersInfo()
+
+	peerClassification := fmt.Sprintf("intrashard:%d,crossshard:%d,unknown:%d,",
+		len(peersInfo.IntraShardPeers),
+		len(peersInfo.CrossShardPeers),
+		len(peersInfo.UnknownPeers),
+	)
+	appStatusHandler.SetStringValue(core.MetricNumConnectedPeersClassification, peerClassification)
+	appStatusHandler.SetStringValue(core.MetricP2pNumConnectedPeersClassification, peerClassification)
+
+	setP2pConnectedPeersMetrics(appStatusHandler, peersInfo)
+	setCurrentP2pNodeAddresses(appStatusHandler, networkComponents)
+}
+
+func setP2pConnectedPeersMetrics(appStatusHandler core.AppStatusHandler, info *p2p.ConnectedPeersInfo) {
+	appStatusHandler.SetStringValue(core.MetricP2pUnknownShardConnectedPeers, sliceToString(info.UnknownPeers))
+	appStatusHandler.SetStringValue(core.MetricP2pIntraShardConnectedPeers, sliceToString(info.IntraShardPeers))
+	appStatusHandler.SetStringValue(core.MetricP2pCrossShardConnectedPeers, sliceToString(info.CrossShardPeers))
+}
+
+func sliceToString(input []string) string {
+	output := ""
+	for _, str := range input {
+		output += str + ","
+	}
+
+	return output
+}
```

Comment:
import (
	"fmt"
	"strings"
)

func main() {
	s := []string{"foo", "bar", "baz"}
	fmt.Println(strings.Join(s, ", "))
}

Reply:
agree, will be done in a next PR as I need 2 more reviewers for this particular change